ckanext/geonetwork/harvesters/utils.py

```diff
@@ -27,46 +27,50 @@ def __init__(self, base, version):
     def retrieveInfo(self, uuid):
 
         if self.version == GEONETWORK_V26:
-            url = "%s/srv/en/mef.export" % self.base
+            # url = "%s/srv/en/mef.export" % self.base
+            url = "%smef.export?uuid=%s" % (self.base, uuid)
+
+            logger.info('URL %r ', url)
+
             #headers = {
-                #"Content-Type": "application/x-www-form-urlencoded",
-                #"Accept": "text/plain"
-            #}
-            query = urllib.urlencode({
-                "uuid": uuid
-            })
+            #    "Content-Type": "application/x-www-form-urlencoded",
+            #    "Accept": "text/plain"
+            
+            #query = urllib.parse.urlencode({
+            #    "uuid": uuid
+            #}).encode('utf-8')
+
+            request = urllib.request.Request(url, method='GET')
 
-            logger.info('Loading MEF for %s', uuid)
-            request = urllib2.Request(url, query)
-            opener = urllib2.build_opener(urllib2.HTTPCookieProcessor(), urllib2.HTTPRedirectHandler())
+            opener = urllib.request.build_opener(urllib.request.HTTPCookieProcessor(), urllib.request.HTTPRedirectHandler())
 
             response = opener.open(request)  # will get a ZIP file
             content = response.read()
-
-            #logger.info('----> %s', content)
+                
             #print 'RESPONSE ', content
 
-            zdata = StringIO(content)
+            zdata = io.BytesIO(content)
             zfile = zipfile.ZipFile(zdata)
 
             xml = None
 
             for name in zfile.namelist():
                 #logger.info(' MEF entry: %s', name)
                 #print ' MEF entry: ', name
-                if name == 'info.xml':
+                if name == 'metadata.xml':
                     uncompressed = zfile.read(name)
                     xml = etree.fromstring(uncompressed)
-
+            
             return xml
 
     def retrieveMetadataCategories(self, uuid):
         xml = self.retrieveInfo(uuid)
-
         cats = []
 
-        for cat in xml.findall('categories/category'):
-            cats.append(cat.get('name'))
+        for cat in xml.iter('{http://www.isotc211.org/2005/gmd}MD_TopicCategoryCode'):
```

Comment:
We need categories from `info.xml`, since the topicCategory in the ISO metadata is a completely different domain.
Categories in GeoNetwork are metadata categorizations outside the metadata schema and related to the GeoNetwork model.